### PR TITLE
Refs Task #402504 test show page dashboard update

### PR DESCRIPTION
**Pull request by nguyenthinhhn**
## Related Tickets

- [#402504](https://dev.sun-asterisk.com/issues/402504)

## What's this PR do ?

- [x] Unittest dashboard

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [x] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [x] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr></table>

**Comment by nguyenthinhhn** [comment]
ready

File: database/factories/BotFactory.php [hunk 1]
```diff
@@ -6,10 +6,11 @@
 use App\Models\User;
 use Faker\Generator as Faker;
 
-$factory->define(Bot::class, function (Faker $faker) {
+$factory->define(Bot::class, function (Faker $faker, $params) {
+    $user_id = (isset($params['user_id'])) ?: factory(User::class)->create()->id;
```
Comment (tamtt-1031) [inline database/factories/BotFactory.php]: vì sao lại đổi thành như thế này e nhỉ?

**Comment by tamtt-1031** [comment]
Em check lại phần phạm vi ảnh hưởng  trong ticket nhé.
a thấy e update lại Factory như vậy mà ko thấy note phạm vi ảnh hưởng liên quan tới việc đổi factory này hết.

File: database/factories/BotFactory.php (hunk 1 above)
Comment (nguyenthinhhn) [reply database/factories/BotFactory.php]: đổi như này thì khi mình tạo bản ghi mình truyền $user_id vào nó sẽ tạo 1 bản ghi có $user_id kia. nếu để như cũ sẽ bị insert thêm những bản ghi khác.

Comment (ghost) [inline database/factories/BotFactory.php]: `(isset($params['user_id'])) ?: factory(User::class)->create()->id;` trả về true khi `isset($params['user_id']) == true`, e check lại nhé

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr></table>

File: database/factories/BotFactory.php (hunk 1 above)
Comment (tamtt-1031) [reply database/factories/BotFactory.php]: a check thì khi để factory cũ. tạo user rồi truyền user_id từ ngoài vào nó vẫn chỉ tạo 1 user.
A nghĩ trường hợp mình truyền từ ngoài vào thì nó sẽ ghi đè lên và ko thực hiện `factory(User::class)->create()->id` 
e kiểm tra lại phát.

**Review by ghost — APPROVED** [review]

**Review by tamtt-1031 — APPROVED** [review]

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr></table>

**Review by ghost — APPROVED** [review]

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**